src/org/opendatakit/briefcase/util/CacheUtils.java

```diff
@@ -0,0 +1,71 @@
+package org.opendatakit.briefcase.util;
+
+import org.opendatakit.briefcase.model.BriefcaseFormDefinition;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.util.HashMap;
+import java.util.Map;
+
+public class CacheUtils {
```

Comment:
Why `CacheUtils` with static methods and not an instantiable `FormCache` or something? Not critical in my view, I just tend to think there are too many “utils” in the world.

Reply:
I think it need to be either singleton or a static utils since it will be shared between classes, right? Yeah, too many utils in here 🕺

Reply:
Did you consider instantiating it once, and passing a reference to where it is needed?

Reply:
I think that will require quiet a bit of changes in method and constructor signature.

We're using this class in the static method `FileSystemUtils.getBriefcaseFormList()`, so we can't just pass it to the `FileSystemUtils`'s constructor. So we need to change the method signature for `getBriefcaseFormList`.

The `getBriefcaseFormList` method is called in 3 different classes, so we need to change the constructor for those 2 of those classes (`PushTransferPanel` and `ExportPanel`) and pass it from the main class `MainBriefcaseWindow`.

What do you think @dcbriccetti?

Reply:
I don’t want to hold you up. I could come along afterward and refactor. But how about adding a static `setFormCache` method to `FileSystemUtils`, and calling it when you instantiate `FormCache`?